src/utils/filterTodos.ts

```diff
@@ -0,0 +1,23 @@
+import { FilterType } from '../types/FilterType';
+import { Todo } from '../types/Todo';
+
+type Queries = {
+  filter: FilterType;
+};
+
+export const filterTodos = (todos: Todo[], { filter }: Queries) => {
+  let filteredTodos = [...todos];
+
+  if (filter !== 'All') {
+    filteredTodos = filteredTodos.filter(todo => {
+      switch (filter) {
+        case 'Active':
+          return !todo.completed;
+        case 'Completed':
+          return todo.completed;
+      }
+    });
+  }
+
+  return filteredTodos;
```

Comment:
```suggestion
  if (filter !== 'All') {
    return todos.filter(todo => {
      switch (filter) {
        case 'Active':
          return !todo.completed;
        case 'Completed':
          return todo.completed;
      }
    });
  }

  return todos;
```
You don't need to create a copy as `filter` doesn't mutate an array